generated_codegen.md

```diff
@@ -568,104 +707,137 @@ Uses of `ExecutableWriterHolder` and `ExecutableWriterHolderNoLog` under `src\co
+```
   2494      size_t dispatchHolderSize = DispatchHolder::GetHolderSize(DispatchStub::e_TYPE_LONG);
   2495      DispatchHolder * holder = (DispatchHolder*) (void*)dispatch_heap->AllocAlignedMem(dispatchHolderSize, CODE_SIZE_ALIGN);
   2496:     ExecutableWriterHolder<DispatchHolder> dispatchWriterHolder(holder, dispatchHolderSize);
   2497
   2498      dispatchWriterHolder.GetRW()->Initialize(holder, addrOfCode,
+```
 
+`VirtualCallStubManager::GenerateResolveStub`
+```
   2602      ResolveHolder * holder = (ResolveHolder*) (void*)
   2603          resolve_heap->AllocAlignedMem(sizeof(ResolveHolder), CODE_SIZE_ALIGN);
   2604:     ExecutableWriterHolder<ResolveHolder> resolveWriterHolder(holder, sizeof(ResolveHolder));
   2605
   2606      resolveWriterHolder.GetRW()->Initialize(holder,
+```
 
+`VirtualCallStubManager::GenerateLookupStub`
+```
   2642      //allocate from the requisite heap and copy the template over it.
   2643      LookupHolder * holder     = (LookupHolder*) (void*) lookup_heap->AllocAlignedMem(sizeof(LookupHolder), CODE_SIZE_ALIGN);
   2644:     ExecutableWriterHolder<LookupHolder> lookupWriterHolder(holder, sizeof(LookupHolder));
   2645
   2646      lookupWriterHolder.GetRW()->Initialize(holder, addrOfResolver, dispatchToken);
-
 ```
 
-`src\coreclr\vm\amd64\cgenamd64.cpp`:
+## `src\coreclr\vm\amd64\cgenamd64.cpp`:
+
+`UMEntryThunkCode::Poison()`
 ```
   599      CONTRACTL_END;
   600
   601:     ExecutableWriterHolder<UMEntryThunkCode> thunkWriterHolder(this, sizeof(UMEntryThunkCode));
   602      UMEntryThunkCode *pThisRW = thunkWriterHolder.GetRW();
   603
+```
 
+Defined in `BEGIN_DYNAMIC_HELPER_EMIT` macro and used in many functions on `DynamicHelpers`.
+```
   769      SIZE_T cbAligned = ALIGN_UP(cb, DYNAMIC_HELPER_ALIGNMENT); \
   770      BYTE * pStartRX = (BYTE *)(void*)pAllocator->GetDynamicHelpersHeap()->AllocAlignedMem(cbAligned, DYNAMIC_HELPER_ALIGNMENT); \
   771:     ExecutableWriterHolder<BYTE> startWriterHolder(pStartRX, cbAligned); \
   772      BYTE * pStart = startWriterHolder.GetRW(); \
   773      size_t rxOffset = pStartRX - pStart; \
+```
 
+`DynamicHelpers::CreateDictionaryLookupHelper` - Called in `prestub.cpp` for `ENCODE_DICTIONARY_LOOKUP_THISOBJ`, `ENCODE_DICTIONARY_LOOKUP_TYPE` and `ENCODE_DICTIONARY_LOOKUP_METHOD`.
+```
   996
   997      GenericHandleArgs * pArgs = (GenericHandleArgs *)(void *)pAllocator->GetDynamicHelpersHeap()->AllocAlignedMem(sizeof(GenericHandleArgs), DYNAMIC_HELPER_ALIGNMENT);
   998:     ExecutableWriterHolder<GenericHandleArgs> argsWriterHolder(pArgs, sizeof(GenericHandleArgs));
   999      argsWriterHolder.GetRW()->dictionaryIndexAndSlot = dictionaryIndexAndSlot;
   1000      argsWriterHolder.GetRW()->signature = pLookup->signature;
-
 ```
 
-`src\coreclr\vm\amd64\jitinterfaceamd64.cpp`:
+## `src\coreclr\vm\amd64\jitinterfaceamd64.cpp`:
+
+`WriteBarrierManager::ChangeWriteBarrierTo` - Alter the `JIT_WriteBarrier`?
```

Comment:
Yes, it changes the write barrier code to a different version (depending on the GC mode etc)